crates/bifrost/src/providers/replicated_loglet/loglet.rs

```diff
@@ -177,6 +177,18 @@ impl<T: TransportConnect> ReplicatedLoglet<T> {
         *self.known_global_tail.get()
     }
 
+    pub async fn shutdown(&self) {
+        if self.sequencer.is_local() {
+            if !self.known_global_tail().is_sealed() {
+                debug!(
+                    loglet_id = %self.my_params.loglet_id,
+                    "Attempting to seal loglet due to shutdown"
+                );
+            }
+            let _ = self.seal().await;
```

Comment:
Why are we unconditionally executing the `seal` call here?

Reply:
The seal call will do nothing if we are already sealed. The check above is just to avoid excessive logging.